components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -1513,82 +1512,39 @@ func socketActivationForDocker(ctx context.Context, wg *sync.WaitGroup, term *te
 	}
 }
 
-func analyseConfigChanges(ctx context.Context, wscfg *Config, w analytics.Writer, cfgobs config.ConfigInterface) {
-	cfgc := cfgobs.Observe(ctx)
-	var (
-		cfg     *gitpod.GitpodConfig
-		t       = time.NewTicker(10 * time.Second)
-		changes []string
-	)
-	defer t.Stop()
-
-	computeHash := func(i interface{}) (string, error) {
-		b, err := json.Marshal(i)
-		if err != nil {
-			return "", err
-		}
-		h := sha256.New()
-		_, err = h.Write(b)
-		if err != nil {
-			return "", err
-		}
-		return fmt.Sprintf("%x", h.Sum(nil)), nil
+func analyseConfigChanges(ctx context.Context, wscfg *Config, w analytics.Writer, cfgobs config.ConfigInterface, gitpodAPI gitpod.APIInterface) {
+	info, err := gitpodAPI.GetWorkspace(ctx, wscfg.WorkspaceID)
```

Comment:
Is there any benefit on calling the API instead of using `os.Getenv("GITPOD_WORKSPACE_ID")`?

Reply:
never mind, I just noticed it's for the owner

Reply:
yes, GITPOD_WORKSPACE_ID is internal implementation, Config abstracts away from it.

Reply:
Yes, @akosyakov always said that once we use `GITPOD_*` env var [[1](https://github.com/gitpod-io/gitpod/pull/8313#discussion_r810007160)] 😂 

But I'm not sure if we have a chance to deprecate it since many users are using Gitpod's env vars. So maybe it's safe somehow? Anyway it must be best if we don't use it

Reply:
there is somethere an issue to provide `gp info` instead and remove all `GITPOD_*` from docs